internal-client/src/main/java/org/apache/uniffle/client/impl/grpc/ShuffleServerGrpcClient.java

```diff
@@ -995,6 +1027,47 @@ public String getClientInfo() {
     return "ShuffleServerGrpcClient for host[" + host + "], port[" + port + "]";
   }
 
+  protected void waitOrThrow(
+      RetryableRequest request, int retry, String requestInfo, StatusCode statusCode, long start) {
+    if (retry >= request.getRetryMax()) {
+      String msg =
+          String.format(
+              "ShuffleServer %s:%s is full when %s due to %s, after %d retries, cost %d ms",
+              host,
+              port,
+              request.operationType(),
+              statusCode,
+              request.getRetryMax(),
+              System.currentTimeMillis() - start);
+      LOG.error(msg);
+      throw new RssFetchFailedException(msg);
+    }
+    try {
+      long backoffTime =
+          Math.min(
+              request.getRetryIntervalMax(),
+              BACK_OFF_BASE * (1L << Math.min(retry, 16)) + random.nextInt(BACK_OFF_BASE));
+      LOG.warn(
+          "Can't acquire buffer for {} from {}:{} when executing {}, due to {}. "
+              + "Will retry {} more time(s) after waiting {} milliseconds.",
+          requestInfo,
+          host,
+          port,
+          request.operationType(),
+          statusCode,
+          request.getRetryMax() - retry,
+          backoffTime);
+      Thread.sleep(backoffTime);
```

Comment:
use wait instead of `sleep`. Because sleep will occupy cpu.

Reply:
I don't think we can optimize this. Here we just wanna sleep for a few seconds and then retry. There is no special condition for `notify` to wake other threads which called `wait`. I think the situation here is not suitable for using `wait`.